ctapipe/image/time_next_neighbor_cleaning.py

```diff
@@ -0,0 +1,742 @@
+            camera. Otherwise the values from a pre filled lookup table are
+            returned
+
+        Returns‚
+        -------
+        comb_factor: int
+            number of possible combinations for camera type and neighbor group
+
+        """
+        # Allow to specify camera and neighbor group by hand. If not set, use
+        # currently selected values.
+        camera_name = camera_name if camera_name else self._camera_name
+        neighbor_group = neighbor_group if neighbor_group else self._neighbor_group
+
+        if not calculate:
+            lookup = dict(
+                LSTCam={"4nn": 75302, "3nn": 19269, "2nn": 5394, "2+1": 71712},
+                ASTRICam={"4nn": 41953, "3nn": 13548, "2nn": 4624, "2+1": 70904},
+                DigiCam={"4nn": 51843, "3nn": 13327, "2nn": 3744, "2+1": 49320},
+                FlashCam={"4nn": 71559, "3nn": 18319, "2nn": 5124, "2+1": 68112},
+                NectarCam={"4nn": 75302, "3nn": 19269, "2nn": 5394, "2+1": 71712},
+                CHEC={"4nn": 36289, "3nn": 11720, "2nn": 4000, "2+1": 61432},
+            )
+            comb_factor = lookup[camera_name][neighbor_group]
+
+        else:
+            from ctapipe.instrument import CameraGeometry
+
+            geometry = CameraGeometry.from_name(camera_name)
+            comb_factor = len(
+                self.get_combinations(
+                    nfold=neighbor_group, mask=geometry, d2=2.4, d1=1.4
+                )
+            )
+            if neighbor_group == "2+1":
+                # current implementation returns 2+1 and 3nn combinations for
+                # 2+1 group search. Need to correct to get correct number.
+                comb_3nn = len(
+                    self.get_combinations(nfold="3nn", mask=geometry, d2=2.4, d1=1.4)
+                )
+                comb_factor -= comb_3nn
+
+        return comb_factor
+
+    def get_time_coincidence(self, charge, accidental_rate, neighbor_group=None):
+        """
+        Get the time coincidence for a given charge (equation 2 of
+        https://arxiv.org/pdf/1307.4939.pdf). The cut on the time coincidence
+        depends on the charge, the multiplicity, combinatorial factor and the
+        accepted accidental rate.
+
+        Parameters
+        ----------
+        charge: numpy.ndarray
+        accidental_rate: astropy.Quantity
+            Allowed accidental rate for thi
+        neighbor_group: string
+            name of neighbor group. Select from 4nn, 3nn, 2+1 and 2nn
+
+        Returns
+        -------
+        time_coincidence: ndarray
+            array of same shape of `charge` filled with the corresponding cut
+            in time
+        """
+        if neighbor_group is not None:
+            self._neighbor_group = neighbor_group
+
+        ipr_graph = interp1d(
+            self._IPR["charge"], self._IPR["rate"], "linear", fill_value="extrapolate"
+        )
+
+        try:
```

Comment:
I think this cannot raise an error anymore, since we now directly use `fill_value="extrapolate"`, so you can remove the `try` and the whole `except` block